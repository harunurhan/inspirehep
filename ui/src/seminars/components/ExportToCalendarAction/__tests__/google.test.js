import { fromJS } from 'immutable';

import getGoogleCalendarUrl from '../google';

describe('google', () => {
  it('matches google url with only required fields', () => {
    const seminar = fromJS({
      title: { title: 'Seminar Title' },
      control_number: 12345,
      start_datetime: '2020-05-15T11:45:00.000000',
      end_datetime: '2020-05-17T00:45:00.000000',
    });

    const url = getGoogleCalendarUrl(seminar);
    expect(url).toMatchSnapshot();
  });

  it('matches google url with all fields', () => {
    const seminar = fromJS({
      title: { title: 'Seminar Title' },
      abstract: {
        value: '<div>This is a <strong>test</strong> seminar<div>',
      },
      control_number: 12345,
      start_datetime: '2020-05-15T11:34:00.000000',
      end_datetime: '2020-05-15T17:34:00.000000',
      address: {
        cities: ['Geneva'],
        place_name: 'CERN',
      },
    });

    const url = getGoogleCalendarUrl(seminar);
    expect(url).toMatchSnapshot();
  });
});
