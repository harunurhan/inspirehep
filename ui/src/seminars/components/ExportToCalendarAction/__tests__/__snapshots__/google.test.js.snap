// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`google matches google url with all fields 1`] = `"https://calendar.google.com/calendar/render?action=TEMPLATE&text=Seminar%20Title&details=This%20is%20a%20test%20seminar&location=CERN&dates=20200515T113400Z%2F20200515T173400Z"`;

exports[`google matches google url with only required fields 1`] = `"https://calendar.google.com/calendar/render?action=TEMPLATE&text=Seminar%20Title&details=&dates=20200515T114500Z%2F20200517T004500Z"`;
